src/Service.Tests/SqlTests/RestApiTests/Put/PutApiTestBase.cs

```diff
@@ -412,6 +412,55 @@ public virtual async Task PutOne_Nulled_Test()
                 );
         }
 
+        /// <summary>
+        /// Test to validate successful execution of a request when a computed field is missing from the request body.
+        /// In such a case, we don't attempt to NULL out computed field (as per PUT semantics) but instead skip updating/inserting the field. 
+        /// </summary>
+        [TestMethod]
+        public virtual async Task PutOneWithComputedFieldMissingFromRequestBody()
+        {
+            // Validate successful execution of a PUT update when a computed field (here 'last_sold_on_update')
+            // is missing from the request body. Successful execution of the PUT request confirms that we did not
+            // attempt to NULL out the 'last_sold_on_update' field.
```

Comment:
I'm confused still. `PutOneUpdateWithComputedFieldMissingFromRequestBody` has this query which checks for a record where the read-only column is null, which contradicts your code comment:
```csharp
            {
                "PutOneUpdateWithComputedFieldMissingFromRequestBody",
                $"SELECT * FROM { _tableWithReadOnlyFields } " +
                $"WHERE [id] = 1 AND [book_name] = 'New book' AND [copies_sold] = 101 AND [last_sold_on] is NULL AND [last_sold_on_date] is NULL " +
                $"FOR JSON PATH, INCLUDE_NULL_VALUES, WITHOUT_ARRAY_WRAPPER"
            },
```

Shouldn't the read-only column be prepopulated with a value and the validation query should be validating that the column has that value (and not NULL)?

Reply:
It looks like you have already made the adjustments for the PUT insert test which proceeds this put-update test.

Reply:
> Shouldn't the read-only column be prepopulated with a value and the validation query should be validating that the column has that value (and not NULL)?

No, it should be NULL. You are referencing a PUT update which does not include last_sold_on in the request body. So that field would be nulled out by DAB. Accordingly the database will null out the last_sold_on_date field as well.

Reply:
We cannot provide any custom value to the last_sold_on field in the request body, because the test is common between the 3 databases, and for MySql there is this bug: https://github.com/Azure/data-api-builder/issues/1696. So, we have to provide a default value of `CURRENT_TIMESTAMP` for the `last_sold_on` field. This value would change everytime the test is re-run and the database is reset.

Reply:
I updated the test to have a value supplied for `last_sold_on` field. DAB would update this field and the database would automatically update the `last_sold_on_date` field